### PR TITLE
Support In-App Safari

### PR DESCRIPTION
**Pull request by divadretlaw**
Add option to open links with In-App-Safari instead of system Safari.

### CODE REVIEW

**Comment by Dimillian** [comment]
It's breaking adding new account for now

**Comment by EvilOne** [comment]
The `StatusCardView` doesn't appear to honour the in-app setting when tapped.

image


https://user-images.githubusercontent.com/169561/211198853-33e0f92a-91e6-48eb-9c4a-2af171b3c1bd.mp4

**Comment by divadretlaw** [comment]
> It's breaking adding new account for now

@Dimillian Using `UIApplication.shared.open(url)` to avoid being captured by In-App Safari and restore previous behavior.

> The StatusCardView doesn't appear to honour the in-app setting when tapped.

@EvilOne Fixed, the SafariRouteur needed to be attached to the `NavigationStack` not the content of it. Otherwise the environment is not propagated to the navigation link destinations.

File: IceCubesApp/App/Tabs/Settings/SettingsTab.swift [hunk 1]
```diff
@@ -82,6 +84,9 @@ struct SettingsTabs: View {
       NavigationLink(destination: remoteLocalTimelinesView) {
         Label("Remote Local Timelines", systemImage: "dot.radiowaves.right")
       }
+      Toggle(isOn: $preferences.useInAppSafari) {
```
Comment (Dimillian) [inline IceCubesApp/App/Tabs/Settings/SettingsTab.swift:87]: Could you make it a Picker instead? It'll make the label shorter and also more clear like "Browser: In App Safari, System Safari" as choices.

Comment (Dimillian) [reply IceCubesApp/App/Tabs/Settings/SettingsTab.swift:87]: On a small iPhone simulator, the label does not fit on one line.

**Comment by EvilOne** [comment]
> > The StatusCardView doesn't appear to honour the in-app setting when tapped.
> 
> Fixed, the SafariRouteur needed to be attached to the `NavigationStack` not the content of it. Otherwise the environment is not propagated to the navigation link destinations.

Moving the `SafariRouteur` to the `NavigationStack` solves the issue and makes total sense! Do we need the two extensions to find the topmost VCs, though? I have the code prior to your changes and just moved the `.withSafariRouteur()` to the `NavigationStack` level, and all worked. I must be missing something.

**Comment by divadretlaw** [comment]
> Moving the `SafariRouteur` to the `NavigationStack` solves the issue and makes total sense! Do we need the two extensions to find the topmost VCs, though? I had the code before your changes and just moved the `.withSafariRouteur()` to the `NavigationStack` level, and all worked. I must be missing something.
> 
> Thanks, and sorry for all the questions.

@EvilOne thank you for testing the PR, questions are always welcome 😄 

Found another issue and resolved it by finding the top-most view controller, since you can only present another view controller from the currently presenting view controller.

The issue was on Account Details, when opening a link within the "About Fields". Since they are opened as a sheet they prevented the `SFSafariViewController` to be presented, because the view controller was no longer the top most one. Now the `SFSafariViewController` should be presented always from the top most view controller.

File: IceCubesApp/App/Tabs/Settings/SettingsTab.swift (hunk 1 above)
Comment (divadretlaw) [reply IceCubesApp/App/Tabs/Settings/SettingsTab.swift]: @Dimillian done, let me know if the implementation of it is ok.

**Comment by EvilOne** [comment]
@divadretlaw Good catch and thanks for the explanation!

**Comment by Dimillian** [comment]
Thanks for all this. I'll take time to do a good round of testing tomorrow before merging.

**Comment by Dimillian** [comment]
Let's have it in now, it seems all good, had little time to test in the end! Thanks a lot!

**Pull request merged**